pkg/dockerregistry/server/util.go

```diff
@@ -40,19 +50,48 @@ func getNamespaceName(resourceName string) (string, string, error) {
 
 // getImportContext loads secrets and returns a context for getting
 // distribution clients to remote repositories.
-func getImportContext(ctx context.Context, secretsGetter secretsGetter, m metrics.Pullthrough) (registryclient.RepositoryRetriever, error) {
+func getImportContext(ctx context.Context, ref *imageapi.DockerImageReference, secrets []corev1.Secret, m metrics.Pullthrough) (registryclient.RepositoryRetriever, error) {
 	req, err := dcontext.GetRequest(ctx)
 	if err != nil {
 		dcontext.GetLogger(ctx).Errorf("unable to get request from context: %v", err)
 		return nil, err
 	}
-	secrets, err := secretsGetter()
+
+	if ref == nil {
+		return nil, fmt.Errorf("invalid nil image reference")
+	}
+
+	if m == nil {
+		return nil, fmt.Errorf("invalid nil metrics pullthrough")
```

Comment:
I don't think it's a good practice to check pointer parameters for nil in every function, sometimes it's better to get a nil pointer dereference panic.

Reply:
I disagree, if a value may be nil functions should be ready to receive it as nil as well.

Reply:
Nevermind, nil checks removed as requested.